exonum-java-binding-core/src/main/java/com/exonum/binding/proxy/NativeHandle.java

```diff
@@ -40,10 +40,10 @@
    * @throws IllegalStateException if this native handle is invalid (nullptr)
    */
   public NativeHandle(long nativeHandle) {
+    checkState(nativeHandle != INVALID_NATIVE_HANDLE, "This handle is not valid: %s", this);
```

Comment:
Do not invoke toString on a partially constructed object

Reply:
Good catch , thanks